### PR TITLE
Initial implementation of ping federate keymanager component

### PR DESCRIPTION
**Pull request by tharindu1st**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

**Review by bhathiya — APPROVED** [review]

File: components/pingfederate.key.manager/src/main/java/org/wso2/pingfederate/client/PingFederateConstants.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * under the License.
+ */
+package org.wso2.pingfederate.client;
+
+/**
+ * This class will hold constants related to PingFederate key manager implementation.
+ */
+public class PingFederateConstants {
+
+    public static final String PING_FEDERATE_TYPE = "Ping-Federate";
```
Comment (bhathiya) [inline components/pingfederate.key.manager/src/main/java/org/wso2/pingfederate/client/PingFederateConstants.java]: shall we make this `PingFederate`?

**Review by Arshardh — APPROVED** [review]

File: components/pingfederate.key.manager/src/main/java/org/wso2/pingfederate/client/PingFederateConnectorConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+    @Override
+    public List<ConfigurationDto> getApplicationConfigurations() {
+
+        List<ConfigurationDto> configurationDtoList = new ArrayList<>();
+//        configurationDtoList
+//                .add(new ConfigurationDto(PingFederateConstants.BYPASS_APPROVAL_PAGES, "Bypass approval Pages",
+//                        "select", "Enable to skip authorization approval pages",
+//                        "false",
+//                        false, false, Arrays.asList("false", "true"), false));
```
Comment (chamilaadhi) [inline components/pingfederate.key.manager/src/main/java/org/wso2/pingfederate/client/PingFederateConnectorConfiguration.java]: shall we remove the commented out code?

File: components/pingfederate.key.manager/src/main/java/org/wso2/pingfederate/model/ClientInfo.java [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+    @SerializedName("clientId")
+    private String clientId;
+    @SerializedName("redirectUris")
+    private Object redirectUris = new ArrayList();
+    @SerializedName("grantTypes")
+    private Object grantTypes = new ArrayList<>();
+    @SerializedName("description")
+    private String description;
+    @SerializedName("clientAuthnType")
+    private String clientAuthnType;
```
Comment (chamilaadhi) [inline components/pingfederate.key.manager/src/main/java/org/wso2/pingfederate/model/ClientInfo.java:42]: is there a meaning for the letter 'n' in  "Authn" ? Is this coming from ping side or a syntax issue in our side?

File: components/pingfederate.key.manager/src/main/java/org/wso2/pingfederate/model/PingFederateDCRClient.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+    @RequestLine("POST")
+    @Headers("Content-Type: application/json")
+    public Response createApplication(ClientInfoList clientInfo);
+
+    @RequestLine("GET /{clientId}")
+    @Headers("Content-Type: application/json")
+    public ClientInfo getApplication(@Param("clientId") String clientId);
+
+    @RequestLine("PUT ")
```
Comment (chamilaadhi) [inline components/pingfederate.key.manager/src/main/java/org/wso2/pingfederate/model/PingFederateDCRClient.java:38]: Don't we have to provide the {clientId} for update

File: components/pingfederate.key.manager/src/main/java/org/wso2/pingfederate/model/ClientInfo.java (hunk 3 above)
Comment (tharindu1st) [reply components/pingfederate.key.manager/src/main/java/org/wso2/pingfederate/model/ClientInfo.java:42]: yes this is from ping end

File: components/pingfederate.key.manager/src/main/java/org/wso2/pingfederate/model/PingFederateDCRClient.java (hunk 4 above)
Comment (tharindu1st) [reply components/pingfederate.key.manager/src/main/java/org/wso2/pingfederate/model/PingFederateDCRClient.java:38]: no pings rest api was not in that mode

**Review by chamilaadhi — APPROVED** [review]

**Pull request merged**